### PR TITLE
raw deployment HPA support

### PR DESCRIPTION
**Pull request by Iamlovingit**
# Raw Deployment autoscaler HPA Support
### Support autoscaler class
* HPA
> Note: HPA is default autoscaler

### Support Metrics
* CPU Utilization
> Note: default CPU Utilization is 80%

### Customized Utilization Value

change value by annotation , e.g.
```
annotations:
"serving.kserve.io/autoscalerClass": "hpa"
"serving.kserve.io/metrics": "cpu"
"serving.kserve.io/targetUtilizationPercentage": "80"
```

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2021 kserve.io.
```
Comment (yuzisun) [inline pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go:2]:
```suggestion
Copyright 2021 The KServe Authors.
```

Comment (yuzisun) [reply pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go:2]: please see https://github.com/kserve/kserve/blob/master/hack/boilerplate.go.txt

Comment (Iamlovingit) [reply pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go]: OK, will follow it soon.

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -68,6 +68,8 @@ var (
 	InferenceServiceGKEAcceleratorAnnotationKey = KServeAPIGroupName + "/gke-accelerator"
 	DeploymentMode                              = KServeAPIGroupName + "/deploymentMode"
 	EnableRoutingTagAnnotationKey               = KServeAPIGroupName + "/enable-tag-routing"
+	HPACPUUtilization                           = KServeAPIGroupName + "/hpa-cpu-utilization"
+	HPAMemoryUtilization                        = KServeAPIGroupName + "/hpa-memory-utilization"
```
Comment (yuzisun) [inline pkg/constants/constants.go:72]: we can follow knative way to separate the type and utilization value, we might want to add other types in future https://knative.dev/docs/serving/autoscaling/concurrency/#target-utilization
https://knative.dev/docs/serving/autoscaling/autoscaling-metrics/
```
serving.kserve.io/metrics: cpu
serving.kserve.io/targetUtilizationPercentage: 80
```

Comment (Iamlovingit) [reply pkg/constants/constants.go]: ok

**Comment by yuzisun** [comment]
moved to kserve repo

**Pull request closed without merging**